### PR TITLE
Add connection caching for mfa and id token

### PR DESCRIPTION
**Pull request by SimbaGithub**
### Description
Add a new authenticator USERNAME_PASSWORD_MFA. 
CLIENT_REQUEST_MFA_TOKEN parameter is added.
Support both ID_TOKEN and MFATOKEN in Local Secure Storage.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by sfc-gh-igarish** [comment]
Too many files to review and conflicting files.

File: secure_storage_manager_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+		if sc.cfg.MfaToken != expectedMfaToken {
+			t.Fatalf("Expected mfa token %v but got %v", expectedMfaToken, sc.cfg.MfaToken)
+		}
+
+		// delete credential and check it no longer exists
+		deleteCredential(sc, mfaToken)
+		getCredential(sc, mfaToken)
+		if sc.cfg.MfaToken != "" {
+			t.Fatalf("Expected mfa token to be empty but got %v", sc.cfg.MfaToken)
```
Comment (sfc-gh-igarish) [inline secure_storage_manager_test.go:29]: is it ok to log token?

File: secure_storage_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	} else {
+		fakeMfaToken := "fakeMfaToken"
+		expectedMfaToken := "fakeMfaToken"
+		sc := getDefaultSnowflakeConn()
+		sc.cfg.Host = "testhost"
+		setCredential(sc, mfaToken, fakeMfaToken)
+		getCredential(sc, mfaToken)
+
+		if sc.cfg.MfaToken != expectedMfaToken {
+			t.Fatalf("Expected mfa token %v but got %v", expectedMfaToken, sc.cfg.MfaToken)
```
Comment (sfc-gh-igarish) [inline secure_storage_manager_test.go:22]: Is it ok to log token?

File: secure_storage_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+		if sc.cfg.IDToken != expectedIDToken {
+			t.Fatalf("Expected id token %v but got %v", expectedIDToken, sc.cfg.IDToken)
+		}
+
+		// delete credential and check it no longer exists
+		deleteCredential(sc, idToken)
+		getCredential(sc, idToken)
+		if sc.cfg.IDToken != "" {
+			t.Fatalf("Expected id token to be empty but got %v", sc.cfg.IDToken)
```
Comment (sfc-gh-igarish) [inline secure_storage_manager_test.go:53]: is it ok to log token?

**Review by sfc-gh-igarish — CHANGES_REQUESTED** [review]
Please check ok to log token or not. I believe it's a sensitive info.

File: secure_storage_manager_test.go (hunk 2 above)
Comment (sfc-gh-ext-simba-lb) [reply secure_storage_manager_test.go:22]: This is just for tests, we are logging a hardcoded fake token "fakeMfaToken".

File: secure_storage_manager_test.go (hunk 1 above)
Comment (sfc-gh-ext-simba-lb) [reply secure_storage_manager_test.go:29]: Same as above. We are logging "fakeMfaToken" for the test

File: secure_storage_manager_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	} else {
+		fakeIDToken := "fakeIDToken"
+		expectedIDToken := "fakeIDToken"
+		sc := getDefaultSnowflakeConn()
+		sc.cfg.Host = "testhost"
+		setCredential(sc, idToken, fakeIDToken)
+		getCredential(sc, idToken)
+
+		if sc.cfg.IDToken != expectedIDToken {
+			t.Fatalf("Expected id token %v but got %v", expectedIDToken, sc.cfg.IDToken)
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lb) [reply secure_storage_manager_test.go:46]: Same as above. Logging fakeIDToken for the test.

File: secure_storage_manager_test.go (hunk 3 above)
Comment (sfc-gh-ext-simba-lb) [reply secure_storage_manager_test.go:53]: Same as above. Logging fakeIDToken for the test.

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM.

**Pull request merged**